src/Artsy/Router/RouterLink.tsx

```diff
@@ -41,8 +42,22 @@ export const RouterLink: React.FC<LinkProps> = ({ to, children, ...props }) => {
       ...handlers,
     ])
 
+    // Wrap any provided onClick with pageview tracking
+    // if the page type stays the same. When linking across apps
+    // those components will get mounted with their own tracking.
+    // But when linking within the same app (just different params),
+    // we want to trigger a pageview.
+    const onClick = e => {
+      const toPath = to.toString()
+      const currentPageType = window.location.pathname.split("/")[1]
+      const toPageType = toPath.split("/")[1]
+
+      if (currentPageType === toPageType) trackPageView({ path: toPath })
```

Comment:
Another minor nit (feel free to ignore if you feel strongly otherwise) 

```js
if (currentPageType === toPageType) {
  trackPageView({ path: toPath })
}

if (props.onClick) {
  props.onClick(event)
}
```
one-liner if's are prone to bugs over time and are harder to read than common multiline ifs.

Reply:
Cool, yea I like this better.